src/sentry/static/sentry/app/views/alerts/rules/details/header.tsx

```diff
@@ -0,0 +1,290 @@
+import React from 'react';
+import {Params} from 'react-router/lib/Router';
+import isPropValid from '@emotion/is-prop-valid';
+import styled from '@emotion/styled';
+import moment from 'moment';
+
+import Breadcrumbs from 'app/components/breadcrumbs';
+import Count from 'app/components/count';
+import DropdownControl from 'app/components/dropdownControl';
+import Duration from 'app/components/duration';
+import ProjectBadge from 'app/components/idBadge/projectBadge';
+import LoadingError from 'app/components/loadingError';
+import MenuItem from 'app/components/menuItem';
+import PageHeading from 'app/components/pageHeading';
+import Placeholder from 'app/components/placeholder';
+import SubscribeButton from 'app/components/subscribeButton';
+import {IconCheckmark} from 'app/icons';
+import {t} from 'app/locale';
+import {PageHeader} from 'app/styles/organization';
+import space from 'app/styles/space';
+import {use24Hours} from 'app/utils/dates';
+import getDynamicText from 'app/utils/getDynamicText';
+import Projects from 'app/utils/projects';
+import {Dataset} from 'app/views/settings/incidentRules/types';
+
+import Status from '../../status';
+import {Incident, IncidentStats} from '../../types';
+import {isOpen} from '../../utils';
+
+type Props = {
+  className?: string;
+  hasIncidentDetailsError: boolean;
+  incident?: Incident;
+  stats?: IncidentStats;
+  onSubscriptionChange: (event: React.MouseEvent) => void;
+  onStatusChange: (eventKey: any) => void;
+  params: Params;
+};
+
+export default class DetailsHeader extends React.Component<Props> {
+  renderStatus() {
+    const {incident, onStatusChange} = this.props;
+
+    const isIncidentOpen = incident && isOpen(incident);
+    const statusLabel = incident ? <StyledStatus incident={incident} /> : null;
+
+    return (
+      <DropdownControl
+        data-test-id="status-dropdown"
+        label={statusLabel}
+        alignRight
+        blendWithActor={false}
+        buttonProps={{
+          size: 'small',
+          disabled: !incident || !isIncidentOpen,
+          hideBottomBorder: false,
+        }}
+      >
+        <StatusMenuItem isActive>
+          {incident && <Status disableIconColor incident={incident} />}
+        </StatusMenuItem>
+        <StatusMenuItem onSelect={onStatusChange}>
+          <IconCheckmark color="green300" />
+          {t('Resolved')}
+        </StatusMenuItem>
+      </DropdownControl>
+    );
+  }
+
+  render() {
+    const {
+      hasIncidentDetailsError,
+      incident,
+      params,
+      stats,
+      onSubscriptionChange,
+    } = this.props;
+    const isIncidentReady = !!incident && !hasIncidentDetailsError;
+    // ex - Wed, May 27, 2020 11:09 AM
+    const dateFormat = use24Hours() ? 'ddd, MMM D, YYYY HH:mm' : 'llll';
+    const dateStarted =
+      incident && moment(new Date(incident.dateStarted)).format(dateFormat);
```

Comment:
if we're using datetime to format we can reuse dateStarted moment object to get the duration

Reply:
I had to do it a bit differently but these will likely change in the next pr in this epic.